lib/lighthouse/veterans_health/client.rb

```diff
@@ -0,0 +1,118 @@
+# one-to-one for each veteran 526 claim for increase submitted in va.gov
+# If you're looking for a more generic Veterans Health Lighthouse client that doesn't instantiate
+# with a set ICN, consider creating/using another client
+module Lighthouse
+  module VeteransHealth
+    # Documentation located at:
+    # https://developer.va.gov/explore/health/docs/fhir?version=current
+    class Client < Common::Client::Base
+      include Common::Client::Concerns::Monitoring
+      configuration Lighthouse::VeteransHealth::Configuration
+
+      # Initializes the Veterans Health client.
+      #
+      # @example
+      #
+      # Lighthouse::VeteransHealth::Client.new(12345)
+      #
+      # @param [String] icn The ICN of the veteran filing the 526 claim for increase
+      #
+      # @return [Lighthouse::VeteransHealth::Client]
+      def initialize(icn)
+        @icn = icn
+      end
+
+      # Handles the Lighthouse request for the passed-in resource.
+      #
+      # @example
+      #
+      # get_resource('conditions')
+      #
+      # @param [String] resource The Lighthouse resource being requested
+      #
+      # @return Faraday::Env response
+      def get_resource(resource)
+        resource = resource.downcase
+        unless %w[conditions medications observations].include?(resource)
+          raise ArgumentError, 'unsupported resource type'
+        end
+
+        send("get_#{resource}")
+      end
+
+      private
+
+      def get_conditions
+        params = {
+          patient: @icn,
+          'clinical-status': 'http://terminology.hl7.org/CodeSystem/condition-clinical|active'
+        }
+
+        perform_get('services/fhir/v0/r4/Condition', params)
+      end
+
+      def get_observations
+        params = {
+          patient: @icn,
+          category: 'vital-signs',
+          code: '85354-9'
+        }
+
+        perform_get('services/fhir/v0/r4/Observation', params)
+      end
+
+      def get_medications
+        params = {
+          patient: @icn
+        }
+        perform_get('services/fhir/v0/r4/MedicationRequest', params)
+      end
+
+      def perform_get(uri_path, params)
+        perform(:get, uri_path, params, headers_hash)
+      end
+
+      def authenticate(params)
+        perform(
+          :post,
+          'oauth2/health/system/v1/token',
+          URI.encode_www_form(params),
+          { 'Content-Type': 'application/x-www-form-urlencoded' }
+        )
+      end
+
+      def bearer_token
+        @bearer_token ||= retrieve_bearer_token
+      end
+
+      def headers_hash
+        @headers_hash ||= Configuration.base_request_headers.merge({ Authorization: "Bearer #{bearer_token}" })
+      end
+
+      def retrieve_bearer_token
+        authenticate_as_system(JwtWrapper.new.token)
+      end
+
+      def authenticate_as_system(json_web_token)
+        authenticate(payload(json_web_token)).body.with_indifferent_access['access_token']
```

Comment:
Do we need with_indifferent_access here?

Reply:
Definitely not, I just had it there because of the test being written to avoid rocket hash syntax. Changed here: https://github.com/department-of-veterans-affairs/vets-api/pull/8558/commits/f45fd6b8253dbcb98eb3611e51b117b90b08a775